recipes-bsp/firmware-imx/firmware-imx_8.6.bb

```diff
@@ -0,0 +1,118 @@
+# Copyright (C) 2012-2016 Freescale Semiconductor
+# Copyright 2017-2019 NXP
+# Copyright (C) 2018 O.S. Systems Software LTDA.
+SUMMARY = "Freescale i.MX firmware"
+DESCRIPTION = "Freescale i.MX firmware such as for the VPU"
+
+require firmware-imx-${PV}.inc
+
+SRC_URI_append = " \
+    file://sdma \
+    file://epdc \
+    file://regulatory \
+    file://hdmi \
+    file://sdma-firmware.service \
+    file://epdc-firmware.service \
+    file://regulatory-firmware.service \
+    file://hdmi-firmware.service \
+"
+
+PE = "1"
+
+inherit allarch systemd
+
+SYSTEMD_PACKAGES = "${PN}-sdma ${PN}-epdc ${PN}-regulatory ${PN}-hdmi"
+SYSTEMD_SERVICE_${PN}-sdma = "sdma-firmware.service"
+SYSTEMD_SERVICE_${PN}-epdc = "epdc-firmware.service"
+SYSTEMD_SERVICE_${PN}-regulatory = "regulatory-firmware.service"
+SYSTEMD_SERVICE_${PN}-hdmi = "hdmi-firmware.service"
+
+do_install() {
+    install -d ${D}${base_libdir}/firmware/imx
+
+    if ${@bb.utils.contains('DISTRO_FEATURES','systemd','true','false',d)}; then
+        install -d ${D}${sysconfdir}
+        install -d ${D}${systemd_system_unitdir}
+        install -m 0755 ${WORKDIR}/sdma ${D}${sysconfdir}
+        install -m 0755 ${WORKDIR}/epdc ${D}${sysconfdir}
+        install -m 0755 ${WORKDIR}/regulatory ${D}${sysconfdir}
+        install -m 0755 ${WORKDIR}/hdmi ${D}${sysconfdir}
+        install -m 0644 ${WORKDIR}/sdma-firmware.service ${D}${systemd_system_unitdir}
+        install -m 0644 ${WORKDIR}/epdc-firmware.service ${D}${systemd_system_unitdir}
+        install -m 0644 ${WORKDIR}/regulatory-firmware.service ${D}${systemd_system_unitdir}
+        install -m 0644 ${WORKDIR}/hdmi-firmware.service ${D}${systemd_system_unitdir}
+    fi
+
+    cd firmware
+    for d in *; do
+        case $d in
+        ddr|seco)
+            # These folders are for i.MX 8 and are included in the boot image via imx-boot
+            bbnote Excluding folder $d
+            ;;
+        *)
+            cp -rfv $d ${D}${base_libdir}/firmware
+            ;;
+        esac
+    done
+    cd -
+
+    # Install SDMA Firmware: sdma-imx6q.bin & sdma-imx7d.bin into lib/firmware/imx/sdma
```

Comment:
Not sure, but are those FW packages needed for *i.MX8M* family?

Reply:
Yes, one of those bins
- sdma-imx7d.bin for i.MX7D/i.MX8M use
- sdma-imx6q.bin for all i.MX6

Reply:
> Yes, one of those bins
> 
>     * sdma-imx7d.bin for i.MX7D/i.MX8M use
> 
>     * sdma-imx6q.bin for all i.MX6

Can they be made machine-dependent here? Having i.MX6 FW file on i.MX8 could be confusing, visa-versa might be even dangerous. 😄

Reply:
Can the name of the sdma-imx7d.bin be changed to also indicate that it is applicable to imx8m? I have run into multiple spots over the last few weeks where trying to figure out if something is applicable to my particular machine is impossible because of naming being specific to an older part but the new parts also use those resources.

Reply:
In worse case we could add an RPROVIDES to it, and use the imx8m package name.

Reply:
> Can the name of the sdma-imx7d.bin be changed to also indicate that it is applicable to imx8m? I have run into multiple spots over the last few weeks where trying to figure out if something is applicable to my particular machine is impossible because of naming being specific to an older part but the new parts also use those resources.

I know it's confusing, the firmware guys don't change names too often.
In a way it makes sense considering that the imx8M family is derived from imx7d, and the sdma IP is exactly the same as imx7d.

Reply:
> I know it's confusing, the firmware guys don't change names too often.
> In a way it makes sense considering that the imx8M family is derived from imx7d, and the sdma IP is exactly the same as imx7d.

For some background here, I have only ever used the imx8m parts and finding out which bits of the 8m IP are the same as previous generations and which are new or changed is difficult. As a newcomer to the i.MX series this has been really confusing to me and I have not found any sort of documentation that compares the different generations as to what changed and what did not.